### PR TITLE
ARROW-13751: Recipe for searching for values matching a predicate

### PR DESCRIPTION
**Pull request by amol-**
https://issues.apache.org/jira/browse/ARROW-13751

### CODE REVIEW

File: python/source/data.rst [hunk 1]
```diff
@@ -137,3 +137,57 @@ function
 .. testoutput::
 
   0 .. 198
+
+Searching for values matching a predicate in Arrays
+---------------------------------------------------
+
+If you have to look for values matching a predicate in Arrow arrays
+the :mod:`arrow.compute` module provides a bunch of predicates that
+can be used to find the values you are looking for.
```
Comment (westonpace) [inline python/source/data.rst:146]:
```suggestion
If you have to look for values matching a predicate in Arrow arrays
the :mod:`arrow.compute` module provides several methods that
can be used to find the values you are looking for.
```

"a bunch of" is a little informal.  Function might be a better word than method, I'm not sure.  I think we should distinguish between "predicate" (the condition we want satisfied) and "method/function" (the tool we use to figure out what satisfies that condition).  A method/function can be a predicate but I think it should return a bool (and not an array of bool) in that case.  For example `lambda x: x > 10` is a predicate but `filter(lambda x: x > 10, my_list)` is not.

File: python/source/data.rst [hunk 2]
```diff
@@ -137,3 +137,57 @@ function
...
+
+For example, given and array with numbers from 0 to 9, if we
```
Comment (westonpace) [inline python/source/data.rst:148]:
```suggestion
For example, given an array with numbers from 0 to 9, if we
```

File: python/source/data.rst [hunk 3]
```diff
@@ -137,3 +137,57 @@ function
...
+want to look only for those greater than 5 we could use the
+func:`arrow.compute.greater` predicate and get back the elements
```
Comment (westonpace) [inline python/source/data.rst:150]:
```suggestion
func:`arrow.compute.greater` method and get back the elements
```

File: python/source/data.rst [hunk 4]
```diff
@@ -137,3 +137,57 @@ function
...
+    false,
+    false,
+    false,
+    true,
+    true,
+    true,
+    true
+  ]
+
+Further more we can filter the array to get only the entries
```
Comment (westonpace) [inline python/source/data.rst:178]:
```suggestion
Furthermore we can filter the array to get only the entries
```

**Review by westonpace — COMMENTED** [review]
Looks good.

**Comment by amol-** [comment]
resolved conflict with main

**Pull request merged**